### DISCUSSION TITLE
Mudblazor Theme?  aka How to have a non-bootstrap based Themes?

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#3014] [General]
We standardized on MudBlazor for our apps we build using Oqtane.  This is causing an interesting situation in 4.0 and beyond where we are constrained to one theme per site.  My challenge is that the Oqtane admin UI is bootstrap, so those UI components require a theme based on bootstrap.   I feel like I'm missing something fundamental.  Will I have to re-write all the OOTB Oqtane admin UI and maintain a parallel UI based on MudBlazor?

**Comment by sbwalker** [comment]
@markdav-is the change which was made in 4.0.0 to only allow a single theme per site could definitely be rolled back. I believe this would solve your problem. However the challenge is what (if anything) can be done in scenarios where someone tries to use 2 different Bootstrap-based themes in the same site - as it will result in JavaScript errors and CSS issues and will lead to the conclusion that Oqtane is "broken". There were already issues logged in this repo which were caused by this scenario - which is why the change was made in 4.0. 

@leigh-pointer had an earlier suggestion to display a message in the Page Settings UI if the Theme chosen is not related to the Default Theme specified in Site Settings ie. "Please Be Aware That Overriding The Default Site Theme With An Unrelated Page Theme May Result In Compatibility Issues For Your Site".  Is this a better solution? Or will users be confused and still log issues?

**Reply by leigh-pointer** [reply]
@markdav-is @sbwalker the functionality to display different themes still work in the framework! Currently the framework filters themes! You could create a module that modifies the page theme yourself and store that in the database! Now i know this works because i do it myself. Designing and building modules i need this, but the situation as it is now in 4.0 i support.

**Reply by sbwalker** [reply]
@leigh-pointer you are correct that the framework does not prevent people from using different themes on different pages in a site - it only prevents it in the UI. I suppose that instead of forcing a workaround like you are suggesting (ie. creating a module to update the page theme) the UI should allow it as well (with a warning message).

**Reply by sbwalker** [reply]
@markdav-is this was resolved in #3018 which will be in 4.0.1

**Reply by markdav-is** [reply]
Thanks for addressing this.  I'll take a look.  In most of my cases, I leave the admin in the oqtane theme and then build custom theme for the custom app we are building.  the site the users see doesn't have direct admin access, so we save a ton of effort not re-theming the admin.
